modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -122,8 +122,15 @@ final case class Inputs(
       case v: Inputs.Virtual =>
         Iterator(v.content, bytes("\n"))
     }
+    val resourceDirsIt = (resourceDirs.flatMap { dir =>
+      os.walk(dir)
+        .filter(os.isFile(_))
+        .map(filePath => s"$filePath:" + os.read(filePath))
```

Comment:
Some resource files might not be readable as strings. Better read things as binary data, compute a SHA-1 of that data say, and use that here.